### DISCUSSION TITLE
MLX90640 support in micropython

### DISCUSSION THREAD

**Discussion by allegfede** [micropython/micropython#9181] [Libraries & Drivers]
Hello, got this interesting sensor in the lab, but have really hard time getting something more than his i2c address from him :smile: 
Here:
https://github.com/avsg8/MLX90640_micropython
there is some sample code, but without the ability to dump all the libs from the archive to the pico filesystem, is really hard to tell if it works or not (i'm using thonny).
Has anyone tried to bind the melexis sensor to the pico? Do you know of some other "driver" to handle it?
Thanks everyone :smile:

**Comment by peterhinch** [comment]
That does look a nice device (for the benefit of others reading it's a 24*32 passive IR imager). The library is an Adafruit one, written for CircuitPython. Options are to run CP and follow the Adafruit instructions or to adapt their driver for MicroPython. I have gone through this process for a number of devices. It is reasonably straightforward once you've figured out the Adafruit way of doing things and will probably result in a driver which is just a single file rather than a whole ecosystem.

**Comment by allegfede** [comment]
That would be great. A single library to copy-paste and save in the lib
folder of the pico would be fantastic.
But at the moment this is far away my possibilities:-(
I'll prefer to use micropython because seems circuit python at the moment
is still monothread (but really love the way circuit python exposed the
file system) and following bytes&bits tutorial video on YouTube I
discovered the great increase on fps displaying things on tiny lcd, using
multicore and video buffer.

Il gio 1 set 2022, 20:27 Peter Hinch ***@***.***> ha scritto:

**Reply by allegfede** [reply]
Do someone have any guidelines on how to embed adafruit libraries in a monolithic one?

**Comment by allegfede** [comment]
someone could help me in the conversion of the circuit python adafruit library for the melexis sensor to the micropython language?

**Reply by peterhinch** [reply]
@jimmo looked at this. He concluded that the Adafruit driver was inefficient and plans to start from scratch and write a MP driver.

**Reply by allegfede** [reply]
That's a great news! I'll try to ask @jimmo if I can help in some way

**Comment by scruss** [comment]
Couple of other library options - untried, at least until I can find my MLX90640 board:

* [mwerezak / micropython-mlx90640](https://github.com/mwerezak/micropython-mlx90640): seems to rely on uasyncio being baked in. No bad thing, but async anything is beyond my tiny brain
* [vertexi / Thermal-camera](https://github.com/vertexi/Thermal-camera): claims to be MicroPython, still looks a lot like CircuitPython to me
